### PR TITLE
fix "resolved" tracking for nested lists 

### LINKED ISSUE #1095: Host creation with multiple interfaces fails
##### SUMMARY

If I try to deploy a new host with two or more interfaces with the FA host module I get an error

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION

```
ansible 2.9.15
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Nov 16 2020, 22:23:17) [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
```

##### COLLECTION VERSION

```
FAM v1.5.0
```

##### KATELLO/FOREMAN VERSION

```
I tried it with Foreman 1.24.3 / Katello 3.14.1 as well as with Foreman 2.1.3 / Katello 3.16.1
```

##### STEPS TO REPRODUCE

```yaml
- hosts: new_centos7.test.local
  gather_facts: false
  tasks:
    - name: Create VM
      theforeman.foreman.host:
        username: "admin"
        password: "admin"
        server_url: "https://test-foreman.test.local"
        validate_certs: false
        name: "{{ inventory_hostname }}"
        location: "Test"
        organization: "Test"
        provision_method: build
        managed: true
        build: true
        hostgroup: "CentOS 7"
        compute_attributes:
          start: "1"
        interfaces_attributes:
          - type: "interface"
            primary: false
            provision: true
            managed: true
            subnet: "Netz1"
            compute_attributes:
              network: "Netz1"
            identifier: "eth0"
          - type: "interface"
            primary: true
            provision: false
            managed: true
            name: "{{ inventory_hostname }}"
            domain: "test.local"
            subnet: "Netz2"
            compute_attributes:
              network: "Netz2"
            identifier: "eth1"
      delegate_to: "test-foreman.test.local"
```

##### EXPECTED RESULTS

A new Centos 7 VM within Foreman with 2 Interfaces

##### ACTUAL RESULTS

```
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_theforeman.foreman.host_payload_s8ppNI/ansible_theforeman.foreman.host_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/foreman_helper.py", line 94, in inner
    return f(self, *args, **kwargs)
  File "/tmp/ansible_theforeman.foreman.host_payload_s8ppNI/ansible_theforeman.foreman.host_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/foreman_helper.py", line 888, in ensure_entity
    updated_entity = self._create_entity(resource, desired_entity, params, foreman_spec)
  File "/tmp/ansible_theforeman.foreman.host_payload_s8ppNI/ansible_theforeman.foreman.host_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/foreman_helper.py", line 947, in _create_entity
    payload = _flatten_entity(desired_entity, foreman_spec)
  File "/tmp/ansible_theforeman.foreman.host_payload_s8ppNI/ansible_theforeman.foreman.host_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/foreman_helper.py", line 1508, in _flatten_entity
    result[flat_name] = [_flatten_entity(ent, foreman_spec[key]['foreman_spec']) for ent in value]
  File "/tmp/ansible_theforeman.foreman.host_payload_s8ppNI/ansible_theforeman.foreman.host_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/foreman_helper.py", line 1502, in _flatten_entity
    result[flat_name] = value['id']
fatal: [new_centos7.test.local -> 10.10.10.10]: FAILED! => changed=false
  invocation:
    module_args:
      architecture: null
      build: true
      comment: null
      compute_attributes:
        start: '1'
      compute_profile: null
      compute_resource: null
      config_groups: null
      content_source: null
      content_view: null
      domain: null
      enabled: null
      environment: null
      hostgroup: CentOS 7
      image: null
      interfaces_attributes:
      - attached_devices: null
        attached_to: null
        bond_options: null
        compute_attributes:
          network: Netz1
        domain: null
        identifier: eth0
        ip: null
        ip6: null
        mac: null
        managed: true
        mode: null
        mtu: null
        name: null
        password: null
        primary: false
        provider: null
        provision: true
        subnet: Netz1
        subnet6: null
        tag: null
        type: interface
        username: null
        virtual: null
      - attached_devices: null
        attached_to: null
        bond_options: null
        compute_attributes:
          network: Netz2
        domain: testdmz.atix
        identifier: eth1
        ip: null
        ip6: null
        mac: null
        managed: true
        mode: null
        mtu: null
        name: new_centos7.test.local
        password: null
        primary: true
        provider: null
        provision: false
        subnet: Netz2
        subnet6: null
        tag: null
        type: interface
        username: null
        virtual: null
      ip: null
      kickstart_repository: null
      lifecycle_environment: null
      location: Test
      mac: null
      managed: true
      medium: null
      name: new_centos7.test.local
      openscap_proxy: null
      operatingsystem: null
      organization: Test
      owner: null
      owner_group: null
      parameters: null
      password: VALUE_SPECIFIED_IN_NO_LOG_PARAMETER
      provision_method: build
      ptable: null
      puppet_ca_proxy: null
      puppet_proxy: null
      puppetclasses: null
      pxe_loader: null
      realm: null
      root_pass: null
      server_url: https://test-foreman.test.local
      state: present
      subnet: null
      subnet6: null
      username: admin
      validate_certs: false
  msg: 'Failed to ensure entity state: TypeError: string indices must be integers, not str'
```

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: plugins/module_utils/foreman_helper.py [hunk 1]
```diff
@@ -834,10 +834,10 @@ def auto_lookup_nested_entities(self):
             if entity_spec.get('type') in {'nested_list'}:
                 for nested_key, nested_spec in self.foreman_spec[key]['foreman_spec'].items():
                     for item in self.foreman_params.get(key, []):
-                        if (nested_key in item and nested_spec.get('resolve', True) and not nested_spec.get('resolved', False)
+                        if (nested_key in item and nested_spec.get('resolve', True)
+                                and item[nested_key] is not None and not isinstance(item[nested_key], (dict, list))
```
Comment (evgeni) [inline plugins/module_utils/foreman_helper.py:838]: @mdellweg I wonder if we should call this `is_resolved(…)` and use it in the normal tracking too?

**Comment by evgeni** [comment]
I'd like to release this as 1.5.1 as soon it's merged.

File: plugins/module_utils/foreman_helper.py (hunk 1 above)
Comment (mdellweg) [reply plugins/module_utils/foreman_helper.py:838]: So you say: "Never save the `resolved` attribute, and always use the heuristic"?
Sounds good to me.

**Review by mdellweg — APPROVED** [review]

File: plugins/module_utils/foreman_helper.py (hunk 1 above)
Comment (evgeni) [reply plugins/module_utils/foreman_helper.py:838]: Yeah, that was the idea. Let's merge this as is and then refactor afterwards.

**Pull request merged**